host_modules/systemd_service.py

```diff
@@ -48,3 +50,19 @@ def stop_service(self, service):
         if result.returncode:
             msg = result.stderr.decode()
         return result.returncode, msg
+
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='s', out_signature='is')
+    def execute_reboot(self, rebootmethod):
+        if rebootmethod in REBOOTMETHOD_COLD_BOOT_VALUES:
+            cmd = ['reboot']
+        elif rebootmethod in REBOOTMETHOD_HALT_BOOT_VALUES:
+            cmd = ['sudo reboot','-p']
```

Comment:
This means look for a binary named `sudo reboot`, which probably  isn't going to work.

Reply:
@vvolam, @saiarcot895  probably this service is running as root and we don't need `sudo`?

Reply:
@vvolam  please specify the full path for the reboot script

Reply:
@vvolam don't we need to specify which DPU to the reboot command?

Reply:
@prgeor "-p" option specifies for pre-shutdown, not related to NPU or DPU.